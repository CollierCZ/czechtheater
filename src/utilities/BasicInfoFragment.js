import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KenticoCloudItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedHtml
            }
            favicon {
                assets{
                  url
                }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;


