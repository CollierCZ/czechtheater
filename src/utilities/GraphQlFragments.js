import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KontentItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedData {
                html
              }
            }
            favicon {
              value {
                url
              }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;

export const ShowListFragment = graphql`
    fragment ShowListFragment on KontentItemShowSection {
      elements {
        shows {
          linked_items {
            elements {
              main_image {
                value {
                  url
                  description
                }
              }
              name {
                value
              }
              url {
                value
              }
              short_description {
                value
              }
              ticket_link {
                value
              }
              gallery {
                value {
                  url
                  description
                }
              }
            }
          }
        }
      }
    }
`;


