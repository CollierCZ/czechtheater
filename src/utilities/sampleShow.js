export const sampleShow = {
    elements: {
      name: {
        value: "Alquist's revenge"
      },
      description: {
        resolvedHtml: "All the acting"
      },
      main_image: {
        assets: [{
          description: "A cool image",
          url: "https://example.com"
        }]
      },
      ticket_link: {
        value: "https://example.com"
      },
      url: {
        value: "alquists-revenge"
      },
      gallery: {
          assets: [
              {
                  description: "A cool image",
                  url: "https://example.com"
              }
          ]
      }
    }
  }