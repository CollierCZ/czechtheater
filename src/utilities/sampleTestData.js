export const sampleShow = {
  elements: {
    name: {
      value: "Alquist's revenge"
    },
    description: {
      resolvedHtml: "<p>All the acting</p>"
    },
    main_image: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    },
    ticket_link: {
      value: "https://example.com"
    },
    url: {
      value: "alquists-revenge"
    },
    gallery: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    }
  }
}

export const sampleShowNoGallery = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.gallery.assets = []

export const sampleShowNoTicketLink = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.ticket_link.value = ""

export const basicInfoQuery = {
  kenticoCloudItemBasicInfo: {
    elements: {
      name: {
        value: "Czech Theater"
      },
      favicon: {
        assets: [{
          url: "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg"
        }]
      },
      short_description: {
        value: "A great theater."
      }
    }
  }
}