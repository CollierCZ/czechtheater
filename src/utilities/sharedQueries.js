import { graphql } from "gatsby";

export const basicInfoQuery = graphql`
  query BasicInfoQuery {
    kenticoCloudItemBasicInfo {
      elements {
        name {
          value
        }
        short_description {
          value
        }
        favicon {
          assets{
            url
          }
        }
      }
    }
  }
`