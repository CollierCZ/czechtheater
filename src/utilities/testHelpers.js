import { shallow } from "enzyme"

export const simpleShallowRender = (component) => {
  const wrapper = shallow (component)
  expect(wrapper).toMatchSnapshot()
}