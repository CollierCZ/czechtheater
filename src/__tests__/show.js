import React from "react";
import { sampleShow, sampleShowNoGallery } from "../utilities/sampleTestData";
import Show from "../templates/show";
import { simpleShallowRender } from "../utilities/testHelpers"

const showToLoad = {}
showToLoad['kontentItemShow'] = sampleShow

const showToLoadNoGallery = {}
showToLoadNoGallery['kontentItemShow'] = sampleShowNoGallery

describe("A show", () => {
    it("loads correctly with a gallery", () => {
      simpleShallowRender(<Show data={showToLoad} />)
    })
    it("loads correctly with no gallery", () => {
      simpleShallowRender(<Show data={showToLoadNoGallery} />)
    })
})