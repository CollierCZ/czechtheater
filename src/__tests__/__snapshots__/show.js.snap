// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A show loads correctly with a gallery 1`] = `
<Layout>
  <SEO
    description="<p>All the acting</p>"
    image="https://example.com"
    keywords={
      Array [
        "czech",
        "theater",
        "show",
      ]
    }
    title="Alquist's revenge"
  />
  <h1>
    Alquist's revenge
  </h1>
  <div
    className="mainImage"
  >
    <img
      alt="A cool image"
      sizes="(max-width: 400px) 360px,(max-width: 800px) 760px,(max-width: 1200) 1140px"
      src="https://example.com"
      srcSet="https://example.com?w=400&fit=crop 400w, https://example.com?w=800&fit=crop 800w, https://example.com?w=1200&fit=crop 1200w, "
    />
  </div>
  <div
    className="showDescription"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>All the acting in a longer way.</p>",
      }
    }
  />
  <h4
    className="tickets"
  >
    <a
      href="https://example.com"
    >
      Tickets
    </a>
  </h4>
  <Gallery
    images={
      Array [
        Object {
          "description": "A cool image",
          "url": "https://example.com",
        },
      ]
    }
  />
</Layout>
`;

exports[`A show loads correctly with no gallery 1`] = `
<Layout>
  <SEO
    description="<p>All the acting</p>"
    image="https://example.com"
    keywords={
      Array [
        "czech",
        "theater",
        "show",
      ]
    }
    title="Alquist's revenge"
  />
  <h1>
    Alquist's revenge
  </h1>
  <div
    className="mainImage"
  >
    <img
      alt="A cool image"
      sizes="(max-width: 400px) 360px,(max-width: 800px) 760px,(max-width: 1200) 1140px"
      src="https://example.com"
      srcSet="https://example.com?w=400&fit=crop 400w, https://example.com?w=800&fit=crop 800w, https://example.com?w=1200&fit=crop 1200w, "
    />
  </div>
  <div
    className="showDescription"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>All the acting in a longer way.</p>",
      }
    }
  />
  <h4
    className="tickets"
  >
    <a
      href="https://example.com"
    >
      Tickets
    </a>
  </h4>
</Layout>
`;

exports[`A show loads correctly with no tickets 1`] = `
<Layout>
  <SEO
    description="<p>All the acting</p>"
    image="https://example.com"
    keywords={
      Array [
        "czech",
        "theater",
        "show",
      ]
    }
    title="Alquist's revenge"
  />
  <h1>
    Alquist's revenge
  </h1>
  <div
    className="mainImage"
  >
    <img
      alt="A cool image"
      sizes="(max-width: 400px) 360px,(max-width: 800px) 760px,(max-width: 1200) 1140px"
      src="https://example.com"
      srcSet="https://example.com?w=400&fit=crop 400w, https://example.com?w=800&fit=crop 800w, https://example.com?w=1200&fit=crop 1200w, "
    />
  </div>
  <div
    className="showDescription"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>All the acting in a longer way.</p>",
      }
    }
  />
  <Gallery
    images={
      Array [
        Object {
          "description": "A cool image",
          "url": "https://example.com",
        },
      ]
    }
  />
</Layout>
`;
